### PR TITLE
Feature/delete datasets and versions from collections

### PR DESCRIPTION
**Pull request by jondewijones**
⚠️⚠️Before merging, make sure related backend branches are merged ⚠️⚠️

### What

- Fix typo in logging line
- Enable deleting of CMD datasets and versions
- Update undo method to undo deleting of versions when deleting related datasets
- Add methods to check with related version exists in a collection and to get the URL of related version in collection

### How to review

Add content (datasets, versions and older style content) to a collection and test delete functionality

### Who can review

Anyone

### CODE REVIEW

File: src/app/views/collections/details/CollectionDetailsController.jsx [hunk 1]
```diff
@@ -384,19 +397,30 @@ export class CollectionDetailsController extends Component {
     }
 
     handleCollectionPageDeleteClick(deletedPage, state) {
```
Comment (dandwal) [inline src/app/views/collections/details/CollectionDetailsController.jsx]: This function is a monolithic 96 liner. I think this needs to be split up, might be worth creating a 'detailsDeletionController' or something, then taking the sub-functions within this function and some others from this class and adding them to it.

File: src/app/views/collections/details/CollectionDetailsController.jsx [hunk 2]
```diff
@@ -369,12 +370,24 @@ export class CollectionDetailsController extends Component {
...
-            pendingDeletedPages: [...state.pendingDeletedPages].filter(pageURI => {
-                return pageURI !== uri;
+    handleCollectionPageDeleteUndo(deleteTimer, uri, notificationID, pageType) {
+        log.event("undo delete page from collection", log.data({url: uri, type: pageType}));
+        let pendingDeletedPages = this.state.pendingDeletedPages.filter(pageURI => {
+            return pageURI !== uri;
+        });
+        if (pageType === "dataset_details") {
+            const datasetID = uri.split("/")[2];
+            const pendingVersionDeleteURL = collections.getURLForVersionInCollection(datasetID, [...this.props.activeCollection.inProgress, ...this.props.activeCollection.reviewed, ...this.props.activeCollection.complete])
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/views/collections/details/CollectionDetailsController.jsx]: Fixed these, but they all wrap in a decent code editor  ¯\_(ツ)_/¯. 
Screenshot 2019-05-09 at 16 29 42

File: src/app/views/collections/details/CollectionDetails.jsx [hunk 3]
```diff
@@ -147,7 +147,7 @@ export class CollectionDetails extends Component {
                         <p>{this.renderLastEditText(page.lastEdit)}</p>
                     </div>
                     <button className="btn btn--primary" onClick={handleEditClick} type="button">{state === "complete" ?  "Review" : "Edit"}</button>
-                    <button className="btn btn--warning btn--margin-left" onClick={handleDeleteClick} type="button" disabled={page.type === "dataset_details" || page.type === "dataset_version"}>Delete</button>
+                    <button className="btn btn--warning btn--margin-left" onClick={handleDeleteClick} type="button">Delete</button>
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/views/collections/details/CollectionDetails.jsx]: Is this comment about the highlighted line? Seems to be different. If it's this line it's already been shortened by removing logic.

File: src/app/utilities/api-clients/collections.js [hunk 4]
```diff
@@ -73,12 +73,15 @@ export default class collections {
...
     }
 
-    static removeDataset(collectionID, datasetID) {
-        console.log(`/zebedee/collections/${collectionID}/datasets/${datasetID}`);
-        // return http.delete(`/zebedee/collections/${collectionID}/datasets/${datasetID}`, true)
-        //     .then(response => {
-        //         return response;
-        //     });
+    static removeDataset(collectionID, datasetID, collectionContent) {
+        const versionInCollection = this.getVersionsInCollectionByDatasetID(datasetID, collectionContent);
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/utilities/api-clients/collections.js]: FIxed

File: src/app/utilities/api-clients/collections.js [hunk 5]
```diff
@@ -100,11 +103,7 @@ export default class collections {
     }
 
     static removeDatasetVersion(collectionID, datasetID, editionID, versionID) {
-        console.log(`/zebedee/collections/${collectionID}/datasets/${datasetID}/editions/${editionID}/versions/${versionID}`);
-        // return http.delete(`/zebedee/collections/${collectionID}/datasets/${datasetID}/editions/${editionID}/versions/${versionID}`, true)
-        //     .then(response => {
-        //         return response;
-        //     });
+        return http.delete(`/zebedee/collections/${collectionID}/datasets/${datasetID}/editions/${editionID}/versions/${versionID}`, true)
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/utilities/api-clients/collections.js]: Fixed

File: src/app/utilities/api-clients/collections.js [hunk 6]
```diff
@@ -118,4 +117,19 @@ export default class collections {
...
+    static getVersionsInCollectionByDatasetID(datasetID, collectionContent) {
+        return collectionContent.find(page => {
+            return page.type === "dataset_version" && page.datasetID === datasetID;
+        })
+    }
+
+    static getURLForVersionInCollection(datasetID, collectionContent) {
+        const version = this.getVersionsInCollectionByDatasetID(datasetID, collectionContent);
+        if (!version) {
+            return
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/utilities/api-clients/collections.js]: Fixed

File: src/app/utilities/api-clients/collections.js [hunk 7]
```diff
@@ -118,4 +117,19 @@ export default class collections {
             })
     }
 
+    static getVersionsInCollectionByDatasetID(datasetID, collectionContent) {
+        return collectionContent.find(page => {
+            return page.type === "dataset_version" && page.datasetID === datasetID;
+        })
```
[the remark this answers is not in the document]
Comment (jondewijones) [reply src/app/utilities/api-clients/collections.js]: FIxed

**Comment by jondewijones** [comment]
Are you running servies with the right feature flags enabled?
**Zebedee**
`ENABLE_DATASET_IMPORT=true`

**Dataset-API**
`ENABLE_DETACH_DATASET=true`
`ENABLE_PRIVATE_ENDPOINTS=true`

**Pull request merged**